pretrain/openwebtext.py

```diff
@@ -88,12 +88,13 @@ def main(fabric, resume) -> None:
     t0 = time.perf_counter()
     with fabric.init_module(empty_init=True):
         model = GPT(config)
+        # TODO: what to do with this?
         model.apply(model._init_weights)
```

Comment:
Yep, we need to find a way to get rid of this. Without a FakeTensor implementation like torchdistx had, we would never be able to support this. Whatever we need to do here needs to go into the reset_parameters() hooks.

Reply:
I agree. What I would do is that `setup_module` basically calls this: https://github.com/Lightning-AI/lit-gpt/blob/main/xla/utils.py#L23-L29

Reply:
Hmm you mean for other strategies? Not for FSDP because that's done internally.

Reply:
Yes, sorry

Reply:
That would probably work yes!